### PR TITLE
BCSC - Frozen on one screen#4300

### PR DESCRIPTION
**Pull request by nitheesh-aot**
*Issue #:*
https://github.com/bcgov/entity/issues/4300

*Description of changes:*

*Checklist:*
I confirm that below checklist items are addressed in this pull request; (check the boxes applicable)
- [ ] No lint errors
- [ ] No test case failures and proper coverage for all modules/classes
- [ ] Updated the deployment configs for new environment variable(s)
- [ ] Updtaed the postman collection in entity repository (https://github.com/bcgov/entity/tree/master/api-e2e/postman) for e2e tests

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-auth license (Apache 2.0).

### CODE REVIEW

File: auth-web/src/components/auth/mixins/NextPageMixin.vue [hunk 1]
```diff
@@ -81,25 +81,27 @@ export default class NextPageMixin extends Vue {
...
         // Redirect to TOS if no terms accepted
         // for invited users , handle user profile
         // Redirect to create team if no orgs
         // Redirect to dashboard otherwise
         if (!this.userProfile?.userTerms?.isTermsOfUseAccepted) {
-          bceditNextStep = Pages.USER_PROFILE_TERMS
+          bceidNextStep = Pages.USER_PROFILE_TERMS
+        } else if (!this.currentUser?.roles?.includes(Role.AccountHolder)) {
+          bceidNextStep = Pages.CHOOSE_AUTH_METHOD
         } else if (!this.currentOrganization && !this.currentMembership) {
```
Comment (saravanpa-aot) [inline auth-web/src/components/auth/mixins/NextPageMixin.vue:93]: this and above check are same ? checking for membership vs accountholder role?

File: auth-web/src/components/auth/mixins/NextPageMixin.vue [hunk 2]
```diff
@@ -81,25 +81,27 @@ export default class NextPageMixin extends Vue {
...
         return `/${nextStep}`
       case LoginSource.BCEID:
-        let bceditNextStep = '/'
+        let bceidNextStep = '/'
         // Redirect to TOS if no terms accepted
         // for invited users , handle user profile
         // Redirect to create team if no orgs
         // Redirect to dashboard otherwise
         if (!this.userProfile?.userTerms?.isTermsOfUseAccepted) {
-          bceditNextStep = Pages.USER_PROFILE_TERMS
```
Comment (saravanpa-aot) [inline auth-web/src/components/auth/mixins/NextPageMixin.vue:90]: does this has to happen only for bcied? not for bcsc

File: auth-web/src/components/auth/mixins/NextPageMixin.vue (hunk 1 above)
Comment (nitheesh-aot) [reply auth-web/src/components/auth/mixins/NextPageMixin.vue:93]: I belive both are different cases, not 100% sure. will the account_holder role will be removed from backend if the user leaves from all his accounts?

File: auth-web/src/components/auth/mixins/NextPageMixin.vue (hunk 2 above)
Comment (nitheesh-aot) [reply auth-web/src/components/auth/mixins/NextPageMixin.vue:90]: no, bcsc as well, have the same check on that case also

File: auth-web/src/components/auth/mixins/NextPageMixin.vue (hunk 1 above)
Comment (nitheesh-aot) [reply auth-web/src/components/auth/mixins/NextPageMixin.vue:93]: no bcsc as well, have the same check on that case also

File: auth-web/src/components/auth/mixins/NextPageMixin.vue (hunk 2 above)
Comment (saravanpa-aot) [reply auth-web/src/components/auth/mixins/NextPageMixin.vue:90]: somehow it doest show up here

File: auth-web/src/components/auth/mixins/NextPageMixin.vue (hunk 1 above)
Comment (saravanpa-aot) [reply auth-web/src/components/auth/mixins/NextPageMixin.vue:93]: can you confirm with @sumesh-aot ?

File: auth-web/src/components/auth/mixins/NextPageMixin.vue (hunk 2 above)
Comment (nitheesh-aot) [reply auth-web/src/components/auth/mixins/NextPageMixin.vue:90]: may be due to that change was already there before, it only shows the changed portion in the file

File: auth-web/src/components/auth/mixins/NextPageMixin.vue (hunk 1 above)
Comment (saravanpa-aot) [reply auth-web/src/components/auth/mixins/NextPageMixin.vue:93]: did u check this?

Comment (sumesh-aot) [reply auth-web/src/components/auth/mixins/NextPageMixin.vue:93]: If the user doesn't have any active accounts account_holder role won't be present.

**Pull request merged**